File: hooks/scripts/session/notification-audit.mjs
#!/usr/bin/env node
import "../suppress-stderr.mjs";
import { auditLog, passthrough, readStdin } from "../_lib.mjs";

(async () => {
	try {
		const data = await readStdin();
		const message = data.message ?? data.notification ?? "";

		auditLog(
			"Notification",
			"notification.mjs",
			"notified",
			message ? String(message).slice(0, 200) : "empty",
		);
		passthrough();
	} catch {
		passthrough();
	}
})();

File: hooks/scripts/session/notification-audit.py
#!/usr/bin/env python3
"""Notification hook: audit logging for notifications.

Logs notification events for enterprise audit trail.
"""

import os
import sys

sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))
from _lib import audit_log, passthrough, read_stdin


def main():
    data = read_stdin()
    message = data.get("message", data.get("notification", ""))

    audit_log(
        "Notification",
        "notification.py",
        "notified",
        reason=str(message)[:200] if message else "empty",
    )
    passthrough()


if __name__ == "__main__":
    main()
